healthcheck/healthcheck_test.go

```diff
@@ -2,95 +2,24 @@ package healthcheck
 
 import (
 	"context"
-	"fmt"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
 	"sync"
 	"testing"
 	"time"
 
+	"github.com/containous/traefik/testhelpers"
 	"github.com/vulcand/oxy/roundrobin"
 )
 
 const healthCheckInterval = 100 * time.Millisecond
 
-type testLoadBalancer struct {
```

Comment:
You could also argue that you need to know the implementation details in order to understand the test.

I don't mind too much either way though, was more interested in the motivation.